includes/core/classes/blocks/class-event-query.php

```diff
@@ -0,0 +1,324 @@
+				$wp_query = new \WP_Query( $filtered_query_args ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+			} else {
+				add_filter(
+					'query_loop_block_query_vars',
+					array( $this, 'query_loop_block_query_vars' ),
+					10,
+					2
+				);
+			}
+		}
+
+		return $pre_render;
+	}
+
+	/**
+	 * Returns an array with Post IDs that should be excluded from the Query.
+	 *
+	 * @param array $attributes Event Query block attributes.
+	 * @return int[] Array of post IDs to exclude.
+	 */
+	protected function get_exclude_ids( array $attributes ): array {
+		$exclude_ids = array();
+
+		// Exclude Current Post.
+		if ( isset( $attributes['exclude_current'] ) && boolval( $attributes['exclude_current'] ) ) {
+			array_push( $exclude_ids, $attributes['exclude_current'] );
+		}
+
+		return $exclude_ids;
+	}
+
+	/**
+	 * Filters the arguments which will be passed to `WP_Query` for the Query Loop Block.
+	 *
+	 * @param array     $query Array containing parameters for <code>WP_Query</code> as parsed by the block context.
+	 * @param \WP_Block $block Block instance.
+	 * @return array Array containing parameters for <code>WP_Query</code> as parsed by the block context.
+	 */
+	public function query_loop_block_query_vars( array $query, \WP_Block $block ): array {
+		// Retrieve the query from the passed block context.
+		$block_query = $block->context['query'];
+
+		// Generate a new custom query with all potential query vars.
+		$query_args = array();
+
+		// Post Related.
+		$query_args['post_type'] = array( Event::POST_TYPE );
+
+		// Type of event list: 'upcoming' or 'past',
+		// @see wp-content/plugins/gatherpress/includes/core/classes/class-event-query.php .
+		$query_args['gatherpress_events_query'] = $block_query['gatherpress_events_query'];
+
+		// Exclude Posts.
+		$exclude_ids = $this->get_exclude_ids( $block_query );
+		if ( ! empty( $exclude_ids ) ) {
+			$query_args['post__not_in'] = $exclude_ids; // phpcs:ignore WordPressVIPMinimum.Performance.WPQueryParams.PostNotIn_post__not_in
+		}
+
+		if ( isset( $block_query['include_unfinished'] ) ) {
+			$query_args['include_unfinished'] = $block_query['include_unfinished'];
+		}
+
+		// Date queries.
+		if ( ! empty( $block_query['date_query'] ) ) {
+			$date_query        = $block_query['date_query'];
+			$date_relationship = $date_query['relation'];
+			$date_is_inclusive = $date_query['inclusive'];
+			$date_primary      = $date_query['date_primary'];
+			$date_secondary    = ! empty( $date_query['date_secondary'] ) ? $date_query['date_secondary'] : '';
+
+			// Date format: 2022-12-27T11:14:21.
+			$primary_year    = substr( $date_primary, 0, 4 );
+			$primary_month   = substr( $date_primary, 5, 2 );
+			$primary_day     = substr( $date_primary, 8, 2 );
+			$secondary_year  = substr( $date_secondary, 0, 4 );
+			$secondary_month = substr( $date_secondary, 5, 2 );
+			$secondary_day   = substr( $date_secondary, 8, 2 );
```

Comment:
I don't love the use of `substr` here. I'd rather us use the `date` function to get the correct formatting from the provided date. Will be easier to understand and less prone to bugs.